### DISCUSSION TITLE
Could DataGridCellsPresenter be unsealed?

### DISCUSSION THREAD

**Discussion by liwuqingxin** [AvaloniaUI/Avalonia#11433] [Q&A]
I would like to implement right frozen column for DataGrid just like it did in wpf [here](https://github.com/dotnet/wpf/issues/2591#issuecomment-990929620). So could the `DataGridCellsPresenter` be unsealed in feature?

**Comment by maxkatz6** [comment]
There is no guarantee that even unsealed it will work just like in WPF. 
Since it's different controls. 

Unsealing it might be a low impact, but without any guarantees.

**Reply by liwuqingxin** [reply]
Thank you for replay. I have implemented the function of freezing column at right with unsealed DataGridCellsPresenter in source code of Avalonia.Controls.DataGrid. I will post it here later.

I have found that DataGrid have less openness than other controls. I can not inherit DataGridRow since it is  just constructed in datagrid. I can not override the DataGridCellsPresenter since it is sealed. It has little ability to customize. As a developer, I expect more scalability and flexibility for a control. Emmmmmmm.... I think the developers should be held accountable for their wildly custom behavior, if they can.

Whether the outcome is open or not, I would be very grateful if you could take this into consideration. 😄
